tc_aws/loaders/presigning_loader.py

```diff
@@ -0,0 +1,39 @@
+# coding: utf-8
+
+from boto.s3.bucket import Bucket
+from thumbor.utils import logger
+from tornado.concurrent import return_future
+
+import thumbor.loaders.http_loader as http_loader
+
+from tc_aws.loaders import *
+from tc_aws.aws.connection import get_connection
+
+
+def _generate_presigned_url(context, bucket, key):
+    connection = get_connection(context)
+    expiry = 60 * 60
+    presigned_url = connection.generate_url(
+        expiry,
+        'GET',
+        bucket,
+        key,
+    )
+    return presigned_url
+
+
+@return_future
+def load(context, url, callback):
+    load_sync(context, url, callback)
+
+def load_sync(context, url, callback):
+    if _use_http_loader(context, url):
```

Comment:
no it won't. our url-generation logic is separated from thumbor and we will have to deploy it separately. so if we can't deploy this addon without changing our urls, this will be a blocker for us to switch.